speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -44,6 +44,16 @@ class SecuredMainPage extends Component {
     }
 
     const { match } = this.props;
+    
+    const { data, error } = this.state;
```

Comment:
Where does this `error` come from?

Also, can you instead set the `error` variable in the `App.js` state (probably have to use a callback)? Please take a look at that file, which fields it expects in the error object. Doing this will automatically show the error page that we already have. No need to create a new div here.

Reply:
I can see there is already an error variable declared in the states of App.js. should i just remove it from securedMainPage?

Reply:
I don't really understand where this `error` variable in the SecuredMainPage comes from. When does it get set?

Reply:
But basically whichever error you are trying to show occurred when you set this variable: set the variable in App.js instead, so that the default error page will get rendered automatically

Reply:
i have no idea why we have this in this page. the variables are not even present when state is declared